wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,381 @@
+    ) from e
+
+
+class WandbLogger:
+    """Log OpenAI fine-tunes to [Weights & Biases](https://wandb.me/openai-docs)."""
+
+    _wandb_api: wandb.Api = None
+    _logged_in: bool = False
+    openai_client: OpenAI = None
+    _run: Run = None
+
+    @classmethod
+    def sync(
+        cls,
+        fine_tune_job_id: Optional[str] = None,
+        openai_client: Optional[OpenAI] = None,
+        first_n_fine_tunes: Optional[int] = None,
+        project: str = "OpenAI-Fine-Tune",
+        entity: Optional[str] = None,
+        overwrite: bool = False,
+        wait_for_job_success: bool = True,
+        **kwargs_wandb_init: Dict[str, Any],
+    ) -> str:
+        """Sync fine-tunes to Weights & Biases.
+
+        :param fine_tune_job_id: The id of the fine-tune (optional)
+        :param openai_client: Pass the `OpenAI()` client (optional)
+        :param first_n_fine_tunes: Number of most recent fine-tunes to log when an fine_tune_job_id is not provided. By default, every fine-tune is synced.
+        :param project: Name of the project where you're sending runs. By default, it is "GPT-3".
+        :param entity: Username or team name where you're sending runs. By default, your default entity is used, which is usually your username.
+        :param overwrite: Forces logging and overwrite existing wandb run of the same fine-tune.
+        :param wait_for_job_success: Waits for the fine-tune to be complete and then log metrics to W&B. By default, it is True.
+        """
+        if openai_client is None:
+            openai_client = OpenAI(
+                api_key=os.environ["OPENAI_API_KEY"],
+            )
+        cls.openai_client = openai_client
+
+        if fine_tune_job_id:
+            wandb.termlog("Retrieving fine-tune job...")
+            fine_tune = openai_client.fine_tuning.jobs.retrieve(
+                fine_tuning_job_id=fine_tune_job_id
+            )
+            fine_tunes = [fine_tune]
+        else:
+            # get list of fine_tune to log
+            fine_tunes = openai_client.fine_tuning.jobs.list()
+            if not fine_tunes or fine_tunes.data is None:
+                wandb.termwarn("No fine-tune has been retrieved")
+                return
+            # Select the `first_n_fine_tunes` from the `fine_tunes.data` list.
+            # If `first_n_fine_tunes` is None, it selects all items in the list (from start to end).
+            # If for example, `first_n_fine_tunes` is 5, it selects the last 5 items in the list.
+            # Note that the last items in the list are the latest fine-tune jobs.
+            fine_tunes = fine_tunes.data[
+                -first_n_fine_tunes if first_n_fine_tunes is not None else None :
+            ]
+
+        # log starting from oldest fine_tune
+        show_individual_warnings = not (
+            fine_tune_job_id is None and first_n_fine_tunes is None
+        )
```

Comment:
i think the simplified version would be more readable ie.:
```suggestion
        show_individual_warnings =  fine_tune_job_id is not None or first_n_fine_tunes is not None
```